File: ABC/abc146/src/bin/c.rs
use proconio::input;

fn main() {
    input! {
        a: usize,
        b: usize,
        x: usize,
    }

    let mut ok = 0;
    let mut no = 1_000_000_001;
    while ok + 1 != no {
        let mid = (ok + no) / 2;
        if check(mid, a, b, x) {
            ok = mid;
        } else {
            no = mid;
        }
    }

    println!("{}", ok);
}

fn check(n: usize, a: usize, b: usize, x: usize) -> bool {
    let l = n.to_string().len();
    a * n + b * l <= x
}

File: ABC/abc146/c.py
a, b, x = map(int, input().split())


def is_ok(n):
    return a * n + b * len(str(n)) <= x


ok, ng = 0, 10**9 + 1
while ng - ok > 1:
    mid = (ok + ng) // 2
    if is_ok(mid):
        ok = mid
    else:
        ng = mid

print(ok)
